### ISSUE TITLE
Direct Link to Customizer Missing in Sites using FSE themes

### ISSUE THREAD

**Issue by druesome** [woothemes/woocommerce#34430]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

It seems that we've hidden the link to the Customizer on sites using FSE/block themes through this PR:

https://github.com/woocommerce/woocommerce/pull/34022

On Atomic sites, the link is already hidden by default on sites using FSE/block themes.  However, for sites that are using WooCommerce, it's a bit problematic because a lot of WooCommerce settings are still tucked within the Customizer, such as:

1. Store Notice
2. Product Catalog
3. Product Images
4. Checkout

This support doc specifically asks users to go to the Customizer to access these settings:

https://woocommerce.com/document/woocommerce-customizer/

There doesn't seem to be an alternative way of changing these settings without visiting the Customizer.  It would be very helpful to have WooCommerce specifically bring back the link until these settings are moved to a different location that doesn't rely on the Customizer.

For now, a workaround would be to direct the user to click Appearance > Additional CSS, then click the back button before they can see the WooCommerce settings.

Another workaround is to install this code snippet which will bring back the link to the Customizer:

```
function add_admin_customizer_fse() {
    add_theme_page( 'Customize', 'Customize', 'manage_options', 'customize.php' );
}
add_action( 'admin_menu', 'add_admin_customizer_fse' );
```

### Expected behavior

I'm able to click the link to the Customizer via Appearance > Customize

### Actual behavior

The link is missing.  I'd have to click Appearance > Additional CSS (then click back) or install a code snippet.

### Steps to reproduce

1. On an Atomic site, activate an FSE theme (ie Twentytwentytwo)
2. Install and activate WooCommerce
3. Change the Store Notice setting as per https://woocommerce.com/document/woocommerce-customizer/

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://andrewscakeshop.wpcomstaging.com
Site address (URL): https://andrewscakeshop.wpcomstaging.com
WC Version: 6.8.0
REST API Version: ✔ 6.8.0
WC Blocks Version: ✔ 8.0.0
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.0.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 7.4.30
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.83.1
OpenSSL/1.1.1n

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.4.25-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.8.0
WC Database Prefix: wp_
Total Database Size: 9.45MB
Database Data Size: 7.86MB
Database Index Size: 1.59MB
wp_woocommerce_sessions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.13MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_ms_snippets: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 2.22MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 2.14MB + Index: 0.06MB + Engine InnoDB
wp_posts: Data: 2.48MB + Index: 0.06MB + Engine InnoDB
wp_snippets: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_deposits_payment_plans: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_deposits_payment_plans_schedule: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wp_phpmyadmin_extension__errors_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

amp_validated_url: 2
attachment: 40
ct_template: 6
customize_changeset: 22
jetpack_migration: 2
jp_img_sitemap: 5
jp_sitemap: 5
jp_sitemap_master: 5
jp_vid_sitemap: 5
nav_menu_item: 6
page: 52
post: 15
revision: 188
web-story: 1
wp_global_styles: 13

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (11) ###

Akismet Anti-Spam: by Automattic – 5.0
All-in-One WP Migration: by ServMask – 7.63
Bradmax Player: by kostalski – 1.1.20
CoBlocks: by GoDaddy – 2.24.2
Code Snippets: by Code Snippets Pro – 3.2.0
WordPress.com Editing Toolkit: by Automattic – 3.39221
Jetpack: by Automattic – 11.3-a.7
Layout Grid: by Automattic – 1.8.2
Reset Global Styles: by Reactor Team – 1.0
WooCommerce: by Automattic – 6.8.1
WP phpMyAdmin: by Puvox.software – 5.2.0.7

### Inactive Plugins (13) ###

AMP: by AMP Project Contributors – 2.3.0
Classic Editor: by WordPress Contributors – 1.6.2
Crowdsignal Forms: by Automattic – 1.6.5
Crowdsignal Polls & Ratings: by Automattic
Inc. – 3.0.9

Disable XML-RPC: by LittleBizzy – 1.0.0
Gutenberg: by Gutenberg Team – 13.9.0
Oxygen: by Soflyy – 3.8
Page Optimize: by Automattic – 0.5.2
Staff of Ra: by  –
Web Stories: by Google – 1.24.0
WooCommerce Deposits: by WooCommerce – 1.5.6
WooCommerce Product Add-ons: by WooCommerce – 4.1.0
WooCommerce Stripe Gateway: by WooCommerce – 6.6.0

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Must Use Plugins (1) ###

WP.com Site Helper: by  –

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔

### WC Pages ###

Shop base: #811 - /shop/
Cart: #157 - /cart/
Checkout: #158 - /checkout/
My account: #159 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Two
Version: 1.2-wpcom (update to version 1.2 is available)
Author URL: https://wordpress.org/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
experimental-product-tour
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
wc-pay-promotion
wc-pay-welcome-page
wc-pay-subscriptions-page

Disabled Features: minified-js
settings

Daily Cron: ✔ Next scheduled: 2022-08-24 09:42:17 +00:00
Options: ✔
Notes: 38
Onboarding: skipped

### Action Scheduler ###

Complete: 17
Oldest: 2022-07-25 06:21:30 +0000
Newest: 2022-08-23 09:44:00 +0000

Pending: 1
Oldest: 2022-08-24 09:42:18 +0000
Newest: 2022-08-24 09:42:18 +0000

### Status report information ###

Generated at: 2022-08-23 09:58:29 +00:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by Addison-Stavlo** [comment]
may be duplicate or related - https://github.com/Automattic/wp-calypso/issues/60732

EDIT - maybe just related - but it seems this issue describes a WooCommerce decision to not show customizer settings when FSE is active but have not yet enabled those settings to be managed elsewhere.

**Comment by nerrad** [comment]
Hi there, thanks for leaving the report.

WooCommerce Core no longer registers customizer settings when a block theme is active. You can [read more about the decision here](https://developer.woocommerce.com/2022/08/11/stop-registering-customizer-options-for-block-themes/). 

We didn't introduce alternatives to the settings for when a block theme is active (yet) because:

- The assumption is that folks switching from a classic theme to a block theme would already have those settings configured. If something needs re-adjusted, a workaround could be switching temporarily to a classic theme and making the adjustments (I know it's less than ideal).
- The assumption is that users starting on a block theme wouldn't necessarily miss having those settings available. However, if they are deemed important, can be achieved by activating a classic theme. Also if the merchant uses Cart and Checkout Blocks they _will_ have a replacement for the checkout-related settings. 

We _do_ have plans to make sure these settings have appropriate alternatives in the context of block themes but felt the confusion around multiple ways to edit a site was a greater concern so released this earlier (especially since its **still possible** to access these settings with a classic theme active).

I would appreciate having some impact into the impact of this decision and whether it's greater than anticipated. 

With that said, we've begun investigating ways of preventing Woo from surfacing the customizer menu but still registering the settings and making them available when the customizer is accessed directly.

**Comment by nerrad** [comment]
The code implementing the customizer change is here:

https://github.com/woocommerce/woocommerce/blob/9e891cf47b9516573b2f5a7d2ee30a4e75fb19d1/plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php#L888-L890

For stores impacted by this that want to re-enable the Woo customizer options when a block theme is active, something like this should work (untested):

```php
add_action( 'woocommerce_loaded', function () {
  if ( wc_current_theme_is_fse_theme() ) {
    new WC_Shop_Customizer();
  }
} );
```

**Comment by nerrad** [comment]
Just a note, the changes [implemented here](https://github.com/woocommerce/woocommerce/pull/34022) should only appear on stores running WooCommerce 6.9 (which is currently still only in beta). @druesome, your reported environment is WC 6.8.0 which shouldn't be impacted by what I've been talking about in here. So I don't know what is happening in your case.

**Comment by druesome** [comment]
@nerrad 

>your reported environment is WC 6.8.0 which shouldn't be impacted by what I've been talking about in here. So I don't know what is happening in your case.

Thank you, I didn't realize that change impacted a different version.  My bad about referencing #34022 as I assumed that PR was hiding the link to the Customizer on top of Calypso already hiding it on my Atomic site.  

Although the main ask of this request is to ensure that the link to the Customizer (wp-admin/customize.php) continue to be surfaced under the Appearance menu while the WooCommerce plugin is active, so as to allow users to access the WooCommerce settings within that section.  

I understand that switching to a Classic theme, clicking Additional CSS (then back) and a snippet can be workarounds, but from a support perspective, it's a bit suboptimal as it leaves the user confused as to how to get to the settings while looking at our available [documentation](https://woocommerce.com/document/woocommerce-customizer/) and as to why the link was being hidden.

I'm guessing that when https://github.com/Automattic/wp-calypso/issues/60732 is resolved the link will come back, but if WooCommerce hides it as well, this will continue to be a point of confusion for users.

**Comment by nerrad** [comment]
> I understand that switching to a Classic theme, clicking Additional CSS (then back) and a snippet can be workarounds, but from a support perspective, it's a bit suboptimal as it leaves the user confused as to how to get to the settings while looking at our available [documentation](https://woocommerce.com/document/woocommerce-customizer/) and as to why the link was being hidden.

Yes, that is already understood. It's why:

- I offered a [code solution](https://github.com/woocommerce/woocommerce/issues/34430#issuecomment-1223987260) that environments such as Atomic or WP.com or other hosts can implement in the meantime if WC 6.9 is released before we can implement...
- This [linked PR to this issue](https://github.com/woocommerce/woocommerce/pull/34443) will ensure that the Woo customizer options are _always registered_ (but just not trigger the customizer menu when a block theme is activated).

**Comment by pehaa** [comment]
Hey, I tested the snippet from @nerrad with the `woocommerce_loaded` hook but it does not seem to work. I'd guess that `woocommerce_loaded` is too early. 

On the other hand `woocommerce_init` did the job correctly and enabled WC Customizer for an FSE theme:

``` php
add_action( 'woocommerce_init', function () {
  if ( wc_current_theme_is_fse_theme() ) {
    new WC_Shop_Customizer();
  }
} );
```

Could you please confirm that's the right workaround for the moment (this is HE asking 🙂) Thanks!

**Comment by masperber** [comment]
I had a chat with a customer today where they needed to change WooCommerce settings on an FSE theme. I gave the workaround to use Appearance > Additional CSS, but this is obviously very confusing. 36906413-hc

**Comment by masperber** [comment]
Customer report: They wanted to change the number of columns in WooCommerce and reported that they couldn't figure out how to access WooCommerce settings. I gave the workaround to use Appearance > Additional CSS.

37024678-hc

**Comment by nerrad** [comment]
One idea that has come up in a Slack conversation with @ellenbauer is that when a block theme is activated if there is an active store notice via the customizer, deactivate it (and remove the customizer option).

**Comment by peterfabian** [comment]
Reading the conversation, it seems to me there's nothing to do for the core team here, with future improvements coming from the Site Editing/Blocks teams in the future. The [workaround](https://github.com/woocommerce/woocommerce/issues/34430#issuecomment-1233754502) @pehaa provided should work in the meantime.

I hope it's fine to close this issue. Please let us know if you disagree and we're happy to reopen or open a new issue. Thanks!
